ext/oci8/oci8_statement.c

```diff
@@ -1083,12 +1083,12 @@ int php_oci_bind_post_exec(zval *data)
 				}
 				break;
 		}
-	} else if ((Z_TYPE_P(bind->zval) == IS_TRUE) || (Z_TYPE_P(bind->zval) == IS_FALSE)) {
+	} else if ((Z_TYPE(bind->val) == IS_TRUE) || (Z_TYPE(bind->val) == IS_FALSE)) {
 		/* This convetrsion is done on purpose (ext/oci8 uses LVAL as a temorary value) */
-		if (Z_LVAL_P(bind->zval) == 0)
-			ZVAL_BOOL(bind->zval, FALSE);
-		else if (Z_LVAL_P(bind->zval) == 1)
-			ZVAL_BOOL(bind->zval, TRUE);
+		if (Z_LVAL(bind->val) == 0)
```

Comment:
@dstogov  I think we need to use "zv" here. bind->val is a reference and zv is the actual value deferenced from bind->zval at Line 980. So we need to check the data type of "zv".  Your previous changes in this function also used "zv".